### ISSUE TITLE
Struggle setup SDK

### ISSUE THREAD

**Issue by Maaarek** [UXDivers/Gorilla-Player-Support#180]
Hi,
the Gorilla player is working really good but I would like to use costume controls. I followed already the setup steps on GP.Github. I am doing something wrong but I don't know what. I am using VS2015 and try to setup the SDK on android.

I would like to use the "ControlTemplate" inside my Portable App.xaml. I included the Gorillla.json + Gorilla SDK only on Android.

Android:
Gorilla.json 
![image](https://cloud.githubusercontent.com/assets/14011754/20194130/080c9462-a7f6-11e6-8aac-4e09b9cb4c2c.png)

![image](https://cloud.githubusercontent.com/assets/14011754/20194012/857df950-a7f5-11e6-9371-a5fd3b199889.png)

This is the page which is using ControlTemplate="{StaticResource  TealTemplate}" It is working fine if I am uploading the application on my device.
![image](https://cloud.githubusercontent.com/assets/14011754/20194078/c31660cc-a7f5-11e6-9707-4e869a4009df.png). 

Thank you,

**Comment by LeoHere** [comment]
@Maaarek, sorry for the delay. Gorilla 0.9.1.2 does not support ControlTemplates. That seems to be the issue here. The next release of Gorilla will support them. 

Just to clarify, in your scenario the SDK does not seems to be needed. I mean, If you are just trying to preview control templates you will not need to use the SDK. The SDK is needed only if you need to access no XAML stuff.

How does this affects you?

**Comment by Maaarek** [comment]
The sample data is working now on my device. I would like to use the Costume Renderer from the native plattform. Sample. Circle Button. What  is the right approach for that ?

**Comment by LeoHere** [comment]
Good to know! 

After you setup the SDK there is not much you need to do. If the circle button is a custom control + custom renderer, make sure that:
1. the PCL assembly where the custom control resides is registered as a known assembly (both ```Gorilla.json``` and ```Config``` object will be required)
2. The custom renderer gets exported in the SDK project.

Let me know if you have any issues.

**Comment by Maaarek** [comment]
Not really. I followed your advice on https://github.com/UXDivers/Gorilla-Player-Support/wiki/Gorilla-SDK.
My CustomeRenderer.cs  and part of Android and not a Xaml file. 
**LoadApplication (UXDivers.Gorilla.Droid.Player.CreateApplication(
    ApplicationContext, 
    new UXDivers.Gorilla.Config("Good Gorilla").RegisterAssemblyFromType<UXDivers.Artina.Shared.CircleImage>()));**

What do you mean with Gorilla.json and Config are required ? Do I have to include something in Gorilla.json and config ?

**Comment by LeoHere** [comment]
```UXDivers.Artina.Shared.dll``` (the PCL assembly where the control resides)  should be registered:
1. in the ```UXDivers.Gorilla.Config``` (as you already did it)
2. in the ```Gorilla.json```, as one of the known assemblies

~~~
{
	"knownAssemblies":[
		{ "name": "UXDivers.Artina.Shared" }
	]
}
~~~

Make sure that the custom render's export is included in your Android project. e.g.

~~~
[assembly: ExportRenderer(
      typeof(UXDivers.Artina.Shared.CircleImage), 
      typeof(UXDivers.Artina.Shared.ImageCircleRenderer))]
~~~

Finally run your project with Gorilla SDK embedded, launching the SDK instead of your app.  Now you should be able to preview the CircleImage control.
